drivers/scheduler/k8s/k8s.go

```diff
@@ -2095,6 +2198,64 @@ func (k *K8s) destroyCoreObject(spec interface{}, opts map[string]bool, app *spe
 
 }
 
+func (k *K8s) destroyApiExtensionsObjects(spec interface{}, app *spec.AppSpec) error {
+
+	if obj, ok := spec.(*apiextensionsv1.CustomResourceDefinition); ok {
+		err := k8sApiExtensions.DeleteCRD(obj.Name)
+		if err != nil {
+			return &scheduler.ErrFailedToDestroyApp{
+				App:   app,
+				Cause: fmt.Sprintf("Failed to destroy CRD: %v. Err: %v", obj.Name, err),
+			}
+		} else {
+			log.Infof("[%v] Destroyed CRD: %v", app.Key, obj.Name)
+			return nil
+		}
+	} else if obj, ok := spec.(*apiextensionsv1beta1.CustomResourceDefinition); ok {
+		err := k8sApiExtensions.DeleteCRDV1beta1(obj.Name)
+		if err != nil {
+			return &scheduler.ErrFailedToDestroyApp{
+				App:   app,
+				Cause: fmt.Sprintf("Failed to destroy CRDV1beta1: %v. Err: %v", obj.Name, err),
+			}
+		} else {
+			log.Infof("[%v] Destroyed CRDV1beta1: %v", app.Key, obj.Name)
+			return nil
+		}
+	}
+
+	return nil
+}
+
+func (k *K8s) destroyAdmissionRegistrationObjects(spec interface{}, app *spec.AppSpec) error {
+
+	if obj, ok := spec.(*admissionregistrationv1.ValidatingWebhookConfiguration); ok {
+		err := k8sAdmissionRegistration.DeleteValidatingWebhookConfiguration(obj.Name)
+		if err != nil {
+			return &scheduler.ErrFailedToDestroyApp{
+				App:   app,
+				Cause: fmt.Sprintf("Failed to destroy ValidatingWebhookConfiguration: %v. Err: %v", obj.Name, err),
+			}
+		} else {
+			log.Infof("[%v] Destroyed ValidatingWebhookConfiguration: %v", app.Key, obj.Name)
+			return nil
+		}
+	} else if obj, ok := spec.(admissionregistrationv1beta1.ValidatingWebhookConfiguration); ok {
+		err := k8sAdmissionRegistration.DeleteValidatingWebhookConfigurationV1beta1(obj.Name)
+		if err != nil {
+			return &scheduler.ErrFailedToDestroyApp{
+				App:   app,
+				Cause: fmt.Sprintf("Failed to destroy ValidatingWebhookConfigurationV1beta1: %v. Err: %v", obj.Name, err),
+			}
+		} else {
+			log.Infof("[%v] Destroyed ValidatingWebhookConfigurationV1beta1: %v", app.Key, obj.Name)
+			return nil
+		}
+	}
+
+	return nil
```

Comment:
not return err? because spec does not match *admissionregistrationv1.ValidatingWebhookConfiguration and admissionregistrationv1beta1.ValidatingWebhookConfiguration.

Reply:
That's not how these functions work. You can look at similar function in this file (which is already being used) in order to understand why it's designed that way. (basically, the reflections continue in other functions and the error is caught elsewhere). This is not an error.

✅